attestation-agent/attester/src/tdx/mod.rs

```diff
@@ -9,10 +9,24 @@ use serde::{Deserialize, Serialize};
 use std::path::Path;
 use tdx_attest_rs;
 
+mod az;
+
 const CCEL_PATH: &str = "/sys/firmware/acpi/tables/data/CCEL";
 
 pub fn detect_platform() -> bool {
-    Path::new("/dev/tdx-attest").exists() || Path::new("/dev/tdx-guest").exists()
+    // Upstream kernel exposes a /dev/tdx_guest device but Intel's lib expects
+    // /dev/tdx-guest
+    let paths = [
+        Path::new("/dev/tdx-attest"),
+        Path::new("/dev/tdx-guest"),
+        Path::new("/dev/tdx_guest"),
```

Comment:
Getting back to the cpuid topic one more time ;-) Would it be sufficient to check `TD guest` cpuid bit here or we want to explicitly check the device nodes are present?

Reply:
In this spot we could check the `TD guest` bit instead, i don't mind doing it that way

Reply:
Do you have a pointer to the cpuid TD guest bit? All I can find cpuid leaf 0x21 which returns the "IntelTDX" string after reassembling registers.

Reply:
i'm now using the cpuid leaf 0x21

Reply:
> Do you have a pointer to the cpuid TD guest bit? All I can find cpuid leaf 0x21 which returns the "IntelTDX" string after reassembling registers.

This is what I meant. Sorry for the confusion about that "bit" which was a misleading word.